### PR TITLE
Reputation system by FilCloud

### PR DESCRIPTION
**Pull request by deaswang**

### CODE REVIEW

File: rfp-proposals/rfp-proposal-reputation-system.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  - Users to query and consume information about a miner in question.
+- Sample web application (can be extremely simple) that uses both this service and go-filecoin to perform the intended workflow.
+- Well-documented api for the service conforming to industry best practices. 
+- Tutorial workflow with demo usage. 
+- Use Golang.
+- Well-documented, human-readable codebase.
+
+## Development Roadmap
+
+| Milestone No. | Milestone Description | Funding | Estimated Timeframe |
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-reputation-system.md]: For each of these milestones, can you describe where the price is coming from?

File: rfp-proposals/rfp-proposal-reputation-system.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+| --- | --- | --- | --- |
+| 1 | Finalize scope, workflow design, and architectural design | 10000 | 2 weeks |
+| 2 | Service implementation | 30000 | 6-8 weeks |
+| 3 | Sample web application demonstrating interoperability of go-filecoin and the proposed service | 10000 | 2-4 weeks |
+| 4 | All project deliverables (website, documentation, codebase) | 10000 | 2-3 weeks |
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-reputation-system.md]: Please provide more detail on each of these milestones

File: rfp-proposals/rfp-proposal-reputation-system.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+## Total Budget Requested
+
+$60000
+
+## Maintenance and Upgrade Plans
+
+We will maintenance and upgrade with go-filecoin upgrade.
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-reputation-system.md]: Will this project also work with lotus?

File: rfp-proposals/rfp-proposal-reputation-system.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+    - Number and % (where applicable) of deals with no penalties
+    - Number and % (where applicable) of slashed contracts
+    - Number and % (where applicable) of dropped contracts
+    - Number and % of failed deals (if possible) -- i.e. the number of deals that were attempted but failed, possibly due to this miner having spotty connectivity
+    - % of retrieval requests successfully fulfilled
+    - Number of files retrieved
+    - Lifetime data stored
+    - Lifetime data retrieved
+    - etc
+  - Storage client Feedback.
```
Comment (pooja) [inline rfp-proposals/rfp-proposal-reputation-system.md]: How will storage clients provide this feedback?

File: rfp-proposals/rfp-proposal-reputation-system.md (hunk 3 above)
Comment (deaswang) [reply rfp-proposals/rfp-proposal-reputation-system.md]: yes. the reputation is depend miner who work on chain.

File: rfp-proposals/rfp-proposal-reputation-system.md (hunk 4 above)
Comment (deaswang) [reply rfp-proposals/rfp-proposal-reputation-system.md]: we can get which client store data to the miner from chain and use client private key to sign feedback and verify it.

**Comment by pooja** [comment]
We've finished the wave 1 grant selection process. Can you please email devgrants@filecoin.org so we can share our decision and next steps, if any? In your email, please include a link to your proposal PR so we know who we're talking to!

Thanks,
Filecoin DevGrants team

**Comment by pooja** [comment]
Greetings! Thank you for your application to the Filecoin DevGrants program. We appreciate your proposal!

You have been provisionally selected for a Wave 1 DevGrant, congratulations! Full acceptance of this proposal is pending some conversation with our team about budget and milestones.

Please reach out to the DevGrants team at devgrants@filecoin.org with a link to this PR, and we will follow up with some of our questions.

Thanks,
Filecoin DevGrants team

**Comment by eshon** [comment]
@deaswang - We were wondering if you would consider a reduced budget by 50% for this project for an MVP because the nodes are undergoing active development in the coming months and this is the kind of tool where the real users and their use cases will become more visible as we get closer to Mainnet. We know it is a harder project than most other ones. But this allows us to de-risk funding a big project that needs to be significantly refactored later on, it may be a little early in terms of all the details of how you would architect this (e.g. Retrieval Market will likely be further defined later on) and there may not have immediate users you can do research and testing with right now. Your team can also apply to extend the grant later on if the MVP is successful.

Reach out if you want to discuss this further.

**Comment by deaswang** [comment]
@eshon Thank you for your reply. What is your Email? We can discuss it there.

**Comment by prastut** [comment]
@eshon 

My team didn't apply for the proposal in the first grant as we went through the [recommended team section](https://github.com/filecoin-project/devgrants/blob/master/rfps/rfp-reputation-system.md#recommended-team), we gained awareness on the kind of skills we need to build to be able to deliver this proposal. And we started investing time to build familiarity with the storage workflows on the Filecoin protocols. 

Since this proposal was provisionally selected, we stopped our effort on our end but if need be can restart the effort. After reviewing the conversations above, I have a grasp of what needs to be done at a high-level view as well as can meet the budget expectations, provided we are able to add value from our end. 

Let me know if there is a possibility for my team to apply.

**Comment by eshon** [comment]
@prastut The Wave 1 deadline has passed but you can apply for a Wave 2 grant. FilCloud was also selected because they have a history of good community contributions to the Filecoin node implementations, in alignment with the recommended team section. It helps if you have a track record like that and there are 'Good First Issue' tickets you can try if you know GoLang. Otherwise consider reapplying in Wave 2.

**Comment by prastut** [comment]
@eshon thanks for the info and appreciate the advice. Yup, we plan to build ecosystem history irrespective of this RFP, I just wanted to gain vantage + notify everyone of the possibility. My team is deep-diving into the Filecoin protocol as we speak, and issue 2757 of building Prometheus metric for Storage power (don't want to tag the issue here as Github will link it) is being considered to start our journey in the Filecoin ecosystem.

**Comment by eshon** [comment]
This proposal was not funded in the end.

**Pull request closed without merging**